opsramp/devmgmt.py

```diff
@@ -41,3 +41,46 @@ def run(self, uuid):
 
     def delete(self, uuid):
         return self.api.delete('%s' % uuid)
+
+
+class Discovery(ApiWrapper):
+    def __init__(self, parent):
+        super(Discovery, self).__init__(parent.api, 'policies/discovery')
+
+    def search(self, pattern=''):
+        return self.api.get('/search?name=%s' % pattern)
+
+    def get(self, discoveryProfileId=''):
+        return self.api.get('/%s' % discoveryProfileId)
+
+    def create(self, definition):
```

Comment:
the api doc says that two of the fields are mandatory in the discovery create request struct? would be good to assert that they exist like "assert 'name' in definition" "assert 'credentialType' in definition' if they are actually mandatory. i know you said the discovery docs are misleading though.